### PR TITLE
Service Builder

### LINKED ISSUE #48: Service Builder Pattern
## Is your feature request related to a problem? Please describe

The patron service is a very complex struct and is build at this point by injecting dependencies.
We could use the builder pattern to make the API much more simple, user friendly and intuitive.

## Describe the solution

We could have something like the following:

```go
s,err:=NewBuilder().WithKafkaConsumer(...).WithAmqpConsumer(...).WithRoutes(...).Build()
```

## Additional context

Given the above we could clean up the service, options and others e.g. routes, middlewares might not need to be a field of the service struct.

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

This PR closes #48

- Introduced a builder
- Introduced make file
- Service constructor unexported
- Servicer options not needed and removed

### CODE REVIEW

**Comment by cpapidas** [comment]
I have checked the code, is much cleaner now. I want to test the actual framework right now to see if we have broken something.

File: service.go [hunk 1]
```diff
@@ -4,95 +4,54 @@ import (
 	"context"
```
Comment (drakos74) [inline service.go]: this file has become very thin now ... should we consider putting the builder logic here as well ?

Comment (mantzas) [reply service.go]: we could!

**Comment by mantzas** [comment]
The PR is too old and the changes are too many... I will open a new PR and redo this... @drakos74 @cpapidas @sotirispl  @moukoublen @fpapadopou

**Pull request closed without merging**